Sources/AblyChat/RoomLifecycleManager.swift

```diff
@@ -216,6 +219,10 @@ internal actor DefaultRoomLifecycleManager<Contributor: RoomLifecycleContributor
                 .suspended(error: error)
             case let .suspended(_, error):
                 .suspended(error: error)
+            case let .failedAwaitingStartOfRundownOperation(_, error):
+                .failed(error: error)
+            case let .failedAndPerformingRundownOperation(_, error):
+                .failed(error: error)
```

Comment:
https://github.com/ably/ably-chat-swift/pull/174#issuecomment-2521332125

>It remains FAILED.

It's here between L223 and 225, right?

Reply:
Not sure I understand the question. 

> What state room is between it goes to FAILED and all channels are detached?

the manager goes through the following statuses:

- during the `ATTACH` operation, before the `RUNDOWN` operation starts: `failedAwaitingStartOfRundownOperation`
- once the `RUNDOWN` operation starts: `failedAndPerformingRundownOperation`
- once the `RUNDOWN` operation completes (i.e. all channels have been detached): `failed`

(all of these manager statuses map to the `failed` room status)